subxt/src/custom_values/custom_value_address.rs

```diff
@@ -42,6 +42,26 @@ pub struct StaticAddress<ReturnTy, IsDecodable> {
     phantom: PhantomData<(ReturnTy, IsDecodable)>,
 }
 
+impl<ReturnTy, IsDecodable> Clone for StaticAddress<ReturnTy, IsDecodable> {
```

Comment:
Why is the manual Clone + Debug impls needed?

It should be possible to derive those I think

Reply:
Because Clone/Debug should be possible regardless of some of the generic params, and the default derives assume that they all impl Clone/debug too.

Buttt, I've been avoiding using derivative and maybe I should just use that instead to automate these better!

Reply:
I mean that if you have:

```rust
#[derive(Clone)]
struct Foo<T> {
    val: usize,
    marker: PhantomData<T>
}
```

Then the `Clone` impl will be like:

```rust
impl <T: Clone> Clone for Foo<T> { ... }
```

Even though it's irrelevant whether `T` impls `Clone` or not. So really we want:

```rust
impl <T> Clone for Foo<T> { ... }
```

I just wrote them manually but should probably switch these to using derivative to save some loc :)

Reply:
I switched to using derivative to derive these things instead to get rid of all of the boilerplate!